### ISSUE TITLE
Translate ggplot2's color gradients to plotly's colorscale

### ISSUE THREAD

**Issue by cpsievert** [ropensci/plotly#232]
Scatter traces in plotly will now support a colorscale (known as a gradient in ggplot2), so this example:

```r
qplot(data = iris, x = Sepal.Length, y = Petal.Length, color = Petal.Width)
``` 

![](https://plot.ly/~cpsievert/1201.png)

should use a colorscale like this:

![](https://plot.ly/~etpinard/3730.png)

**Comment by cpsievert** [comment]
For future reference, I've been playing with this on #226, and [have a working version for `plot_ly()`](https://github.com/ropensci/plotly/blob/ed6a586721bc2eb9bc0d3fa5d600b6ed442d971f/R/utils.R#L123-L135) which could fairly easily be brought over to `ggplotly()`

```r
plot_ly(data = iris, x = Sepal.Length, y = Petal.Length, color = Petal.Width, mode = "markers")
```

![](https://plot.ly/~cpsievert/1261.png)

(another factor to consider in all this is that [discretizing continuous color scales can be perceptually advantageous](http://hypertextbook.com/facts/2006/JenniferLeong.shtml)).

**Comment by joshmyzie2** [comment]
Has this been partially implemented?  I noticed it is working (really well) for geom_tile(), but not geom_point() or geom_line().

![untitled](https://cloud.githubusercontent.com/assets/12620937/12948663/5ede7824-cffb-11e5-8f7d-795b773ed944.png)

**Comment by cpsievert** [comment]
This should be fixed in f1900cc3cd464408a60ffadfe3e95460747430e0.
